daemon/runtime_unix.go

```diff
@@ -116,7 +117,10 @@ func (daemon *Daemon) rewriteRuntimePath(name, p string, args []string) (string,
 func (daemon *Daemon) getRuntime(name string) (*types.Runtime, error) {
 	rt := daemon.configStore.GetRuntime(name)
 	if rt == nil {
-		return nil, errdefs.InvalidParameter(errors.Errorf("runtime not found in config: %s", name))
+		if !isPermissibleC8dRuntimeName(name) {
+			return nil, errdefs.InvalidParameter(errors.Errorf("unknown or invalid runtime name: %s", name))
+		}
+		return &types.Runtime{Shim: &types.ShimConfig{Binary: name}}, nil
```

Comment:
This PR is also laying some groundwork for the follow-up PR to enable a containerd runtime to be set as the default runtime. That function is not returning an error directly because the appropriate error to return is context-dependent. A different error message is more appropriate when validating the configured default runtime.

> Of course at that point it's debatable if it should also handle converting it to a `types.Runtime`, so something like;
> 
> ```go
> return getContainerdRuntime(name)
> ```

That hypothetical `getContainerdRuntime` function would be a near drop-in replacement for `(*Daemon).getRuntime`, having the same signature and everything. And it would almost seem to work, too. But it would be subtly broken to call it _anywhere_ other than `(*Daemon).getRuntime` as it would disregard any configured runtimes. I would prefer to have only one function which maps a name to a runtime so there is no possibility to choose wrong.

Reply:
> But it would be subtly broken to call it anywhere other than (*Daemon).getRuntime as it would disregard any configured runtimes. 

Overall I'm not _too_ concerned (for a non-exported function, which more commonly come with some guidance). Was mostly considering it to be nice if it would return a nicely formatted error with the right type ("invalid name"), and generally `err` can be used as a "boolean" for validation.

Not a blocker though.